webapp/channels/src/components/common/hooks/useCWSAvailabilityCheck.ts

```diff
@@ -7,15 +7,20 @@ import {useSelector} from 'react-redux';
 import {Client4} from 'mattermost-redux/client';
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
 
-export default function useCWSAvailabilityCheck() {
-    const [canReachCWS, setCanReachCWS] = useState(false);
+export default function useCWSAvailabilityCheck(teamEditionDefaultAllowed = false) {
+    const [canReachCWS, setCanReachCWS] = useState<boolean | undefined>(undefined);
```

Comment:
Maybe we can leave as boolean type. And then change this check
```js
...
const canReachCWS = useCWSAvailabilityCheck(true);
...

    if (typeof canReachCWS !== 'undefined' && !canReachCWS) {
        return (
            <AirGappedModal
                onClose={handleOnClose}
            />
        );
    }
```

to 
```js
...
const teamEditionAllowedToCheckAvailability = true;
const canReachCWS = useCWSAvailabilityCheck(true);
...

    if (teamEditionAllowedToCheckAvailability && !canReachCWS) {
        return (
            <AirGappedModal
                onClose={handleOnClose}
            />
        );
    }


```

Reply:
Doing it that way would mean on first mount `canReachCWS` is false, so the AirGappedModal would appear until the CWS availability check returned true. With just binary true/false you can't actually determine "is this false because it can't reach CWS, or is it false because it hasn't yet tried to reach CWS?" - `undefined` as the default satisfies the latter case while also being falsey for any existing usages